### ISSUE TITLE
Warrior (and probably other castes) can click through corners that are blocked by walls.

### ISSUE THREAD

**Issue by usnpeepoo** [cmss13-devs/cmss13#925]
## Description
When there are 2 walls on the top right, and bottom left corners of a 4x4 space, and a warrior is on the top left, and a marine on the bottom right, a warrior (or other castes) can use an adjacent click abillity and harm/stun/fling the marine on the opposite section. This should not be possible.

## Difference between expected and actual behavior
Expected: Click doesn't go through as you are not "really" adjacent to the marine (blocked by 2 walls)
Actual: you can click the marine and still use your abillity on them.

## Steps to reproduce
https://streamable.com/lupkj7
recreate the walls and the positions of xeno/marine

## Issue bingo (not required to be all checked)

- [x] Issue could be reproduced at least once
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

**Comment by usnpeepoo** [comment]
what works with this bug: dancer tail trip, vanguard ranged & aoe slash, rav eviscerate, boiler acid lance & gas, trapper acid mines & deploy trap, defender tail whip and probably some other drone stuff

**Comment by Wintermote** [comment]
Acid traps are also triggered like this. They don't check if both diagonals are blocked and thus the diagonal between them is blocked too.

**Comment by usnpeepoo** [comment]
https://github.com/cmss13-devs/cmss13/pull/951
partially fixed by this PR for warrior

**Comment by Huffie56** [comment]
so the warrior part as been fixed so i will closed this and open a more detail and up to date listing everything related that is an issue

**Comment by Huffie56** [comment]
https://github.com/cmss13-devs/cmss13/issues/4077 this is the new issue i will try to add everything that's left on it.
